### PR TITLE
feat: implementation of collector http client with pure hyper

### PR DESCRIPTION
**Pull request by tomkarw**
Closes https://github.com/open-telemetry/opentelemetry-rust/issues/843.

### CODE REVIEW

File: opentelemetry-jaeger/tests/integration_test.rs [hunk 1]
```diff
@@ -110,6 +110,17 @@ mod tests {
                         .expect("cannot create tracer using default configuration")
                 }),
             ),
+            (
+                "collector_hyper",
+                Box::new(|| {
+                    opentelemetry_jaeger::new_collector_pipeline()
+                        .with_endpoint(collector_endpoint)
+                        .with_hyper()
```
Comment (TommyCpp) [inline opentelemetry-jaeger/tests/integration_test.rs:118]: Probably need to add the hyper under `integration_test` features

**Comment by tomkarw** [comment]
Hey @TommyCpp, I have no idea why the `cargo doc` keeps failing. I fixed a couple of the errors (even in unrelated parts of the code base), but this latest one in `opentelemetry-proto` is too much for me. I can't even reproduce it locally (the local docs look fine, and it's a crate that is not pushed publicly).

Could you lend me a hand, or maybe I can just `[allow(clippy::...)]` it?

**Comment by TommyCpp** [comment]
> Hey @TommyCpp, I have no idea why the `cargo doc` keeps failing. I fixed a couple of the errors (even in unrelated parts of the code base), but this latest one in `opentelemetry-proto` is too much for me. I can't even reproduce it locally (the local docs look fine, and it's a crate that is not pushed publicly).
> 
> Could you lend me a hand, or maybe I can just `[allow(clippy::...)]` it?

Thanks! `coverage` uses nightly build so it may have additional lint issue. I won't worry it too much. I will take a look and open a PR to fix it if needed.

**Comment by tomkarw** [comment]
Hey @TommyCpp,

I've added hyper-with-tls support and refactored feature-conditional imports.

I deem this PR ready for review now.

**Review by TommyCpp — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**